src/compiler/checker.ts

```diff
@@ -2766,7 +2766,9 @@ namespace ts {
         }
 
         function getBaseTypeNodeOfClass(type: InterfaceType): ExpressionWithTypeArguments {
-            return getClassExtendsHeritageClauseElement(<ClassLikeDeclaration>type.symbol.valueDeclaration);
+            let classDeclaration =
+                getDeclarationOfKind(type.symbol, SyntaxKind.ClassDeclaration) || type.symbol.valueDeclaration;

```

Comment:
I don't actually see that as a better idea. The `valueDeclaration` is the first value declaration found, I don't think you should be changing how other code uses it as a form of convenience for this one case. Otherwise, you'll have to go through each use of valueDeclaration and check how it's being used.